### PR TITLE
OCM-3155 | feat:Added KmsArnRegex validator + tests

### PR DESCRIPTION
**Pull request by den-rgb**
Added KMS ARN regex validator + tests
https://issues.redhat.com/browse/OCM-3155

Related to:
- https://github.com/terraform-redhat/terraform-provider-rhcs/pull/334
- https://github.com/openshift/rosa/pull/1530

### CODE REVIEW

File: pkg/resource/validations/kmsArnRegexValidation.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+var KmsArnRE = regexp.MustCompile(
+	`^arn:aws[\w-]*:kms:[\w-]+:\d{12}:key\/mrk-[0-9a-f]{32}$|[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`,
+)
+
+func ValidateKMSKeyARN(kmsKeyARN *string) error {
+	if kmsKeyARN != nil {
+		if !KmsArnRE.MatchString(*kmsKeyARN) {
+			return fmt.Errorf("expected a valid value for kms-key-arn matching %s", KmsArnRE)
```
Comment (robpblake) [inline pkg/resource/validations/kmsArnRegexValidation.go:15]: @den-rgb I don't know how useful this error message is to the user :-) Maybe it's just better to do something like:

```
return fmt.Errorf("The KMS Key ARN '%s' is not a valid ARN", kmsKeyARN)
```

I personally think that's much clearer for the user where the problem is rather than them trying to grok the regex and figure out where the problem might be.

WDYT?

File: pkg/resource/validations/kmsArnRegexValidation_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	Describe("validateKMSKeyARN", func() {
+		var (
+			kmsKeyARN  string
+		)
+
+		BeforeEach(func() {
+			kmsKeyARN = ""
+		})
+
+		Context("when kmsKeyARN is nil", func() {
```
Comment (robpblake) [inline pkg/resource/validations/kmsArnRegexValidation_test.go:19]: @den-rgb Does this test pass? From the code I don't see you returning an `error` if the `kmsKeyArn` is `nil`?

Comment (den-rgb) [reply pkg/resource/validations/kmsArnRegexValidation_test.go:19]: It expects an error on line 22, it gets that error from line 21

Comment (den-rgb) [reply pkg/resource/validations/kmsArnRegexValidation_test.go:19]: That test was checking if it was empty not nil, I'll add an extra test case for checking if it's nil and I'll refactor the context

File: pkg/resource/validations/kmsArnRegexValidation.go (hunk 1 above)
Comment (den-rgb) [reply pkg/resource/validations/kmsArnRegexValidation.go:15]: the point of the code is to check if it matches the regexp, so I still think it would be beneficial to include an example of some sort of why it's wrong, so I could maybe add an example arn?

File: pkg/resource/validations/kmsArnRegexValidation.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+func ValidateKMSKeyARN(kmsKeyARN *string) error {
+	if *kmsKeyARN == "" {
+		return fmt.Errorf("expected a non empty value for kms-key-arn")
+	}
+
+	if !KmsArnRE.MatchString(*kmsKeyARN) {
+			return fmt.Errorf("expected the kms-key-arn: %s to match %s",*kmsKeyARN, KmsArnRE)
+		}
```
Comment (robpblake) [inline pkg/resource/validations/kmsArnRegexValidation.go:19]: @den-rgb Minor formatting issue here on the `{` placement.

**Review by robpblake — APPROVED** [review]
LGTM.

File: pkg/resource/validations/kmsArnRegexValidation.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	"fmt"
+	"regexp"
+)
+
+var KmsArnRE = regexp.MustCompile(
+	`^arn:aws[\w-]*:kms:[\w-]+:\d{12}:key\/mrk-[0-9a-f]{32}$|[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`,
+)
+
+func ValidateKMSKeyARN(kmsKeyARN *string) error {
+	if *kmsKeyARN == "" {
```
Comment (ciaranRoche) [inline pkg/resource/validations/kmsArnRegexValidation.go:13]: Can we add a nil check on this before dereferencing it

Comment (ciaranRoche) [reply pkg/resource/validations/kmsArnRegexValidation.go:13]: Also add a test case for nil strings :pray:

Comment (den-rgb) [reply pkg/resource/validations/kmsArnRegexValidation.go:13]: We can but i don't think it'll ever be nil due to it being assigned as a string type
